python/private/py_library.bzl

```
@@ -292,12 +338,17 @@ def _get_venv_symlinks(ctx):
             first_level_explicit_packages.append(d)
 
     for dirname in first_level_explicit_packages:
-        venv_symlinks.append(VenvSymlinkEntry(
+        prefix = dir_symlinks[dirname]
+        entry = VenvSymlinkEntry(
             kind = VenvSymlinkKind.LIB,
-            link_to_path = paths.join(repo_runfiles_dirname, site_packages_root, dirname),
+            link_to_path = paths.join(prefix, site_packages_root, dirname),
+            package = package,
+            version = version_str,
             venv_path = dirname,
-        ))
-    return venv_symlinks
+        )
+        venv_symlinks.append(entry)
+
+    return depset(venv_symlinks)
```

Missing topological order here. Maybe this is the cause of everything?

Depsets with an explicit order will check their compatibility. However, depsets without an explicit ordering will happily merge with whatever. This behavior is shallowly enforced, from what I recall. i.e. you can merge a postorder with preorder depset by wrapping one in a default order. idk what behavior is exactly when this is done.

```suggestion
    return depset(venv_symlinks, order="topological")
```